pkg/api/handlers/compat/containers_create.go

```diff
@@ -262,10 +263,17 @@ func cliOpts(cc handlers.CreateContainerConfig, rtc *config.Config) (*entities.C
 	}
 
 	// special case for NetworkMode, the podman default is slirp4netns for
-	// rootless but for better docker compat we want bridge.
+	// rootless but for better docker compat we want bridge. Do this only we
+	// the default config in containers.conf wasn't overridden to use another
+	// value than the default "private" one.
 	netmode := string(cc.HostConfig.NetworkMode)
+	configDefaultNetNS := util.DefaultContainerConfig().Containers.NetNS
 	if netmode == "" || netmode == "default" {
-		netmode = "bridge"
+		if configDefaultNetNS == "" || configDefaultNetNS == string(specgen.Default) || configDefaultNetNS == string(specgen.Private) {
```

Comment:
I don't think private == bridged. AFAIK private is `net=none`?

Reply:
It's not my understanding, neither of the code, neither of what I am witnessing right now (with a RHEL's 4.2 podman):

```
rgeissler@NCEOBERHEL80009.rnd.amadeus.net[dev]:~$ podman create --rm --network=private fedora
a9ba5324003e4017d4e7b39e5d799d7a3d4c3a78a4a11466096b6310f675b2f5
rgeissler@NCEOBERHEL80009.rnd.amadeus.net[dev]:~$ podman inspect a9ba5324003e4017d4e7b39e5d799d7a3d4c3a78a4a11466096b6310f675b2f5|grep NetworkMode
               "NetworkMode": "slirp4netns",
rgeissler@NCEOBERHEL80009.rnd.amadeus.net[dev]:~$ podman create --rm --network=none fedora
67c53908e0cc6dfe59149e9341a91537faa0782f2c71a83396d782a661a7fb14
rgeissler@NCEOBERHEL80009.rnd.amadeus.net[dev]:~$ podman inspect 67c53908e0cc6dfe59149e9341a91537faa0782f2c71a83396d782a661a7fb14|grep NetworkMode
               "NetworkMode": "none",
rgeissler@NCEOBERHEL80009.rnd.amadeus.net[dev]:~$ podman create --rm fedora
55ee6884b388a174a5bdc572dc4b5de766aceeb0952c5cfeaf299eae8255615a
rgeissler@NCEOBERHEL80009.rnd.amadeus.net[dev]:~$ podman inspect 55ee6884b388a174a5bdc572dc4b5de766aceeb0952c5cfeaf299eae8255615a|grep NetworkMode
               "NetworkMode": "slirp4netns",
```

It seems that "private" is kind of the default mode, which defaults to "slirp4netns" in rootless mode and I think it's "bridge" in rootful mode (didn't check the rootful case). It doesn't seem to be an alias of "none".

In `specgen.ParseNetworkFlag` it seems that all three values (empty string, "default", "private") mean the same thing, that's why I took these 3 strings here as well.

Reply:
That was not the original intent of the code when I wrote it, but if it works like that right now (and it looks like a deliberate change), that sounds like what we'll be going with.

Reply:
To be honest, initially I was tempted to just inconditionnally remove the rewriting to "bridge" if, but the comment on top stopped me from doing this, as it seemed it was really done on purpose to really mimic Docker. I personally care much more about the case I test (ie people using the `quay.io/podman/stable` image who have explictely `netns=host` in their containers.conf and expect it to be respected when using the docker client).